### PR TITLE
Move Tutorials into Documentation

### PR DESCRIPTION
**Pull request by ScalaWilliam**
In my view, tutorials are definitely part of the documentation (eg think
Python: https://docs.python.org/3/tutorial/index.html ).

The structure will be clearer and you can navigate straight from a
tutorial back to the documentation.

To make sure we don't have any broken links, I installed the
jekyll-redirect-from plugin which creates a static redirect.

Deleted nuused 'boost' and 'firstapp' pages: it's better not to have
this sort of thing lying around (and possibly getting indexed by Google,
and when a user visits the page they find nothing).

### CODE REVIEW

**Comment by sjrd** [comment]
@ochrons Do you have any opinion on this reorganization?

File: .htaccess [hunk 1]
```diff
@@ -6,7 +6,11 @@ Update this file if you have an entry that you are moving around.
 Don't update this file if you are adding a new entry.
 {% endcomment %}
 
-redirect 301 /doc/tutorial.html {{ BASE_PATH }}/tutorial/basic/
+redirect 301 /doc/tutorial.html {{ BASE_PATH }}/doc/tutorial/basic/
+redirect 301 /tutorial/basic/ {{ BASE_PATH }}/doc/tutorial/basic/
+redirect 301 /tutorial/ {{ BASE_PATH }}/doc/tutorial/
+redirect 410 /tutorial/boost/
+redirect 410 /tutorial/firstapp/
```
Comment (ScalaWilliam) [inline .htaccess]: I must note that this `.htaccess` is inconsistent, some parts of it use this `BASE_PATH`, others don't.

File: .htaccess [hunk 2]
```diff
@@ -6,7 +6,11 @@ Update this file if you have an entry that you are moving around.
 Don't update this file if you are adding a new entry.
 {% endcomment %}
 
-redirect 301 /doc/tutorial.html {{ BASE_PATH }}/tutorial/basic/
+redirect 301 /doc/tutorial.html {{ BASE_PATH }}/doc/tutorial/basic/
+redirect 301 /tutorial/basic/ {{ BASE_PATH }}/doc/tutorial/basic/
+redirect 301 /tutorial/ {{ BASE_PATH }}/doc/tutorial/
+redirect 410 /tutorial/boost/
```
Comment (ScalaWilliam) [inline .htaccess]: Added `410 Gone`, as I removed these pages in the PR.

**Comment by ScalaWilliam** [comment]
@sjrd I made the respective changes in `.htaccess`.

**Comment by ScalaWilliam** [comment]
@sjrd this is outstanding, just a reminder

File: _config.yml [hunk 3]
```diff
@@ -59,6 +59,6 @@ colors:  #in hex code if not noted else
 versions:
   scalaJS: 0.6.25
   scalaJSBinary: 0.6
-  scalaJSDev: 1.0.0-M5
-  scalaJSDevBinary: 1.0.0-M5
+  scalaJSDev: 1.0.0-M6
+  scalaJSDevBinary: 1.0.0-M6
```
Comment (sjrd) [inline _config.yml]: This change is already on `master`. You probably need a `rebase`.

**Comment by ScalaWilliam** [comment]
@sjrd rebased.

**Pull request merged**